commands/connect/diagnose.js

```diff
@@ -50,18 +56,40 @@ module.exports = {
     context.region = connection.region_url
     let results = yield cli.action('Diagnosing connection', co(function * () {
       let url = '/api/v3/connections/' + connection.id + '/validations'
-      return yield api.request(context, 'GET', url)
+      try {
+        let {data: {result_url: resultUrl}} = yield api.request(context, 'POST', url)
+
+        let i = 0
+
+        while (true) {
+          if (i > 20) {
+            cli.error('There was an issue retrieving validations')
```

Comment:
Does it help to know that the task will fail after ~300s?

Reply:
Assuming that the API sends a 4xx status when that happens, the CLI should throw an error and bail out of this while loop

Reply:
That said, I can adjust the timeout/# of iterations to match 300s so we don't exit early